### PR TITLE
feat(env): implement unified file loading in console-web

### PR DESCRIPTION
**Pull request by ygrishajev**
refs #313

### CODE REVIEW

File: apps/deploy-web/src/instrumentation.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export async function register() {
```
Comment (ygrishajev) [inline apps/deploy-web/src/instrumentation.ts:1]: this is how we can run validation once on server start up

File: apps/deploy-web/next.config.js [hunk 2]
```diff
@@ -10,6 +10,16 @@ const withPWA = require("next-pwa")({
 });
 const { withSentryConfig } = require("@sentry/nextjs");
 
+try {
+  const { browserEnvSchema } = require("./env-config.schema");
```
Comment (ygrishajev) [inline apps/deploy-web/next.config.js:14]: this file is compiled as a part of a `build` npm script. So it is expected to only be present at docker build time. Or anytime the command is executed off

File: apps/deploy-web/src/config/env-config.schema.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import { z } from "zod";
+
+const networkId = z.enum(["mainnet", "sandbox", "testnet"]);
+const coercedBoolean = () => z.enum(["true", "false"]).transform(val => val === "true");
+
+export const browserEnvSchema = z.object({
```
Comment (baktun14) [inline apps/deploy-web/src/config/env-config.schema.ts:6]: I think we're missing `NEXT_PUBLIC_GA_MEASUREMENT_ID` ?

File: apps/deploy-web/src/config/env-config.schema.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+export const serverEnvSchema = browserEnvSchema.extend({
+  MAINTENANCE_MODE: coercedBoolean().optional().default("false"),
+  AUTH0_SECRET: z.string(),
+  AUTH0_BASE_URL: z.string().url(),
+  AUTH0_ISSUER_BASE_URL: z.string().url(),
+  AUTH0_CLIENT_ID: z.string(),
+  AUTH0_CLIENT_SECRET: z.string(),
+  AUTH0_AUDIENCE: z.string(),
+  AUTH0_SCOPE: z.string()
```
Comment (baktun14) [inline apps/deploy-web/src/config/env-config.schema.ts:32]: I have `AUTH0_M2M_DOMAIN`, `AUTH0_M2M_CLIENT_ID` and `AUTH0_M2M_CLIENT_SECRET` locally, do we need them? @Redm4x

File: apps/deploy-web/src/instrumentation.ts (hunk 1 above)
Comment (baktun14) [reply apps/deploy-web/src/instrumentation.ts:1]: ref: https://nextjs.org/docs/app/building-your-application/optimizing/instrumentation

**Review by baktun14 — APPROVED** [review]
Well done!

File: apps/deploy-web/src/config/env-config.schema.ts (hunk 4 above)
Comment (ygrishajev) [reply apps/deploy-web/src/config/env-config.schema.ts:32]: can't find not where they used, nor here https://auth0.github.io/nextjs-auth0/types/config.ConfigParameters.html

are they needed? seems like it all works without them

Comment (ygrishajev) [reply apps/deploy-web/src/config/env-config.schema.ts:32]: wait a min, found them

Comment (ygrishajev) [reply apps/deploy-web/src/config/env-config.schema.ts:32]: but again, they are not set in prod or staging. Are the needed?

**Pull request merged**